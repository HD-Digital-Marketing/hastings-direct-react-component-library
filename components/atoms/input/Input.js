import React from 'react'
import { css } from 'glamor'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './input.scss'

const Input = props => {

    const {
        theme = '',
        type = 'text',
        id = '',
        size = '',
        length = '',
        maxLength = '',
        readOnly = false,
        disabled = false,
        min = '',
        max = '',
        multiple = false,
        required = false,
        step = '',
        autoFocus = false,
        autoComplete = 'on',
        name = '',
        placeholder = '',
        value = '',
        className = '',
        wrapClass = '',
        icon
    } = props

    let inlineCss = css({
        ':before': {
            'background-image': `url(${icon ?? ''})`
        },
    })

    const inputClass = classNames('a-input', className, {
        'a-input--success': theme === 'success',
        'a-input--error': theme === 'error',
        'a-input--reg': theme === 'reg',
        'a-input--error a-input--reg': theme === 'reg-error',
        'a-input--value': theme === 'value',
        'a-input--error a-input--value': theme === 'value-error',
        'a-input--miles': theme === 'miles',
        'a-input--error a-input--miles': theme === 'miles-error',
        'a-input--icon-left': theme === 'icon-left',
        'a-input--error a-input--icon-left': theme === 'icon-left-error',
        'a-input--icon-right': theme === 'icon-right',
        'a-input--error a-input--icon-right': theme === 'icon-right-error'
    })

    const inputWrapClass = classNames('a-input__wrap', {
        'a-input__wrap--success': wrapClass === 'success',
        'a-input__wrap--error': wrapClass === 'error',
        'a-input__wrap--reg': wrapClass === 'reg',
        'a-input__wrap--reg a-input__wrap--error': wrapClass === 'reg-error',
        'a-input__wrap--value': wrapClass === 'value',
        'a-input__wrap--value a-input__wrap--error': wrapClass === 'value-error',
        'a-input__wrap--miles': wrapClass === 'miles',
        'a-input__wrap--miles a-input__wrap--error': wrapClass === 'miles-error',
        'a-input__wrap--icon a-input__wrap--icon-left': wrapClass === 'icon-left',
        'a-input__wrap--icon a-input__wrap--icon-left a-input__wrap--error': wrapClass === 'icon-left-error',
        'a-input__wrap--icon a-input__wrap--icon-right': wrapClass === 'icon-right',
        'a-input__wrap--icon a-input__wrap--icon-right a-input__wrap--error': wrapClass === 'icon-right-error'
    })

    return (
        <div className={inputWrapClass} {...inlineCss}>
            <input {...{
                id,
                size,
                length,
                maxLength,
                readOnly,
                disabled,
                min,
                max,
                multiple,
                required,
                step,
                autoFocus,
                autoComplete,
                type,
                name,
                placeholder }}
                   className={inputClass}
                   defaultValue={value} />
        </div>
    )
}

Input.propTypes = {
    type: PropTypes.string,
    name: PropTypes.string,
    id: PropTypes.string,
    size: PropTypes.string,
    length: PropTypes.string,
    maxLength: PropTypes.string,
    readOnly: PropTypes.bool,
    disabled: PropTypes.bool,
    min: PropTypes.string,
    max: PropTypes.string,
    multiple: PropTypes.bool,
    required: PropTypes.bool,
    step: PropTypes.string,
    autoFocus: PropTypes.bool,
    autoComplete: PropTypes.string,
    placeholder: PropTypes.string,
    value: PropTypes.string,
    className: PropTypes.string,
}

export default Input
