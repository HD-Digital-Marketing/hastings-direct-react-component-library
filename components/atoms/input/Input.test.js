import React from 'react';
import Input from './Input';

describe('Check the input renders', () => {
    test('it should render the input', () => {

        const renderedInput = Input({
            type: 'search',
            value: '1234',
            name: 'testInput',
            placeholder: 'Test text'
        });

        expect(renderedInput).toEqual(<input type="search" name="testInput" className="a-input" placeholder="Test text" defaultValue="1234" />);
    });
});