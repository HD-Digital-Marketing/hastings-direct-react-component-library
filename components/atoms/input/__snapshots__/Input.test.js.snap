// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the alert renders it should match the snapshot 1`] = `
<input
  autoComplete="on"
  autoFocus={false}
  className="a-input"
  defaultValue="123456789"
  disabled={false}
  id="someId"
  length=""
  max=""
  maxLength=""
  min=""
  multiple={false}
  name="someName"
  placeholder="This is placeholder text"
  readOnly={false}
  required={false}
  size=""
  step=""
  type="text"
/>
`;
